hy/completer.hy

```diff
@@ -0,0 +1,111 @@
+(import contextlib)
+(import os)
+(import re)
+(import sys)
+
+(import hy.macros)
+(import hy.compiler)
+(import [hy.lex.parser [mangle unmangle]])
+(import [hy.-compat [builtins string-types]])
+
+(if-python2
+  (import [--builtin-- [eval]])
+  (import [builtins [eval]]))
+
+(setv docomplete True)
+(try
+  (import readline)
+  (except [ImportError]
+    (try
+      (import pyreadline.rlmain)
+      (import pyreadline.unicode-helper)
+      (import readline)
+      (except [ImportError]
+        (setv docomplete False)))))
+
+(defclass Completer []
+  (defn --init-- [self &optional [namespace {}]]
+    (when (not (instance? dict namespace))
+      (raise (TypeError "namespace must be a dictionary")))
+
+    (setv self.namespace namespace
```

Comment:
Shouldn't you copy `namespace` (or set the default argument of `namespace` to `None` and then assign `{}` in the body)?

Reply:
Yeah, good catch. I ported over the old code without thinking hard enough.